fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStore.java

```diff
@@ -1209,6 +1211,97 @@ public void addRecordWriteConflict(@Nonnull Tuple primaryKey) {
+        ExecuteState executeState = scanProperties.getExecuteProperties().getState();
+        // Get the cursor with the index entries and the records from FDB
+        RecordCursor<FDBIndexedRawRecord> indexEntries = indexMaintainer.scanIndexPrefetch(range, hopInfo.pack(), continuation, scanProperties);
+
+        RecordCursor<FDBIndexedRecord<Message>> indexedRecordCursor = indexEntries.mapPipelined(indexedRawRecord -> {
+            // Use the raw record entries to reconstruct the original raw record (include all splits and version, if applicable)
+            FDBRawRecord fdbRawRecord = unsplitSingleRecord(recordSubspace, sizeInfo, indexedRawRecord.getRawRecord(), scanProperties, useOldVersionFormat());
+
+            Optional<CompletableFuture<FDBRecordVersion>> versionFutureOptional = Optional.empty();
+            // The version future will be ignored in case the record already has a version
+            if (useOldVersionFormat() && !fdbRawRecord.hasVersion()) {
+                versionFutureOptional = loadRecordVersionAsync(indexedRawRecord.getIndexEntry().getPrimaryKey());
+            }
+            final ByteScanLimiter byteScanLimiter = executeState.getByteScanLimiter();
+            // TODO: Should we add the index entry bytes to the scanned bytes count?
+            byteScanLimiter.registerScannedBytes(sizeInfo.getKeySize() + sizeInfo.getValueSize());
+
+            // Deserialize the raw record
+            CompletableFuture<FDBStoredRecord<Message>> storedRecord = deserializeRecord(serializer, fdbRawRecord, metaDataProvider.getRecordMetaData(), versionFutureOptional);
+            return storedRecord.thenApply(rec -> new FDBIndexedRecord<>(indexedRawRecord.getIndexEntry(), rec));
+        }, 1);
+
+        return context.instrument(FDBStoreTimer.Events.LOAD_RECORD, indexedRecordCursor);
+    }
+
+    /**
+     * Create and return an instance of {@link FDBRawRecord} from a given {@link MappedKeyValue}. The given
+     * parameter represents a series of record splits (and optional version), that the method will "unsplit" and reconstruct
+     * into a single raw record.
+     * @param recordSubspace the subspace for the record to allow the record keys to be identified from the splits
+     * @param sizeInfo Size Info to collect metrics
+     * @param mappedResult the record splits, packed into a KeyValueAndMappedReqAndResult
+     * @param scanProperties the scam properties to use for the unsplitting
+     * @param oldVersionFormat whether to use the old version record format when reading the records
+     * @return an instance of {@link FDBRawRecord} reconsrtucted from the given record splits
+     */
+    private FDBRawRecord unsplitSingleRecord(final Subspace recordSubspace, final SplitHelper.SizeInfo sizeInfo, final MappedKeyValue mappedResult, final ScanProperties scanProperties, final boolean oldVersionFormat) {
+        List<KeyValue> recordSplits = mappedResult.getRangeResult();
```

Comment:
Can this be empty in the case of a dangling index entry, causing `IllegalResultValueAccessException` below? Do we want an exception? Should this return `null`?

Reply:
Done. Handle empty `rangeResult` by consulting the `orphanBehavior`